### PR TITLE
Fix messaging for consecutive transactions

### PR DESCRIPTION
**Pull request by ThibautBremand**

### CODE REVIEW

File: packages/extension/src/contexts/index.ts [hunk 1]
```diff
@@ -2,4 +2,5 @@ export * from './BrowserContext';
 export * from './LedgerContext';
 export * from './NetworkContext';
 export * from './ServerContext';
+export * from './TransactionProgress';
```
Comment (FlorianBouron) [inline packages/extension/src/contexts/index.ts:5]: Could you please rename it to: TransactionProgressContext please?

File: packages/extension/src/chromeServices/background/index.ts [hunk 2]
```diff
@@ -66,6 +70,30 @@ chrome.runtime.onMessage.addListener((e) => {}); // keepAlive
...
+
+  const openedWindows = await chrome.windows.getAll();
+  const { currentWindowId } = await chrome.storage.local.get('currentWindowId');
+
+  if (currentWindowId && openedWindows.find((window) => window.id === currentWindowId)) {
+    await chrome.storage.local.remove('currentWindowId');
+    await chrome.windows.remove(currentWindowId);
+  }
+
+  saveInChromeSessionStorage('hasTxInProgress', true);
```
Comment (ThibautBremand) [inline packages/extension/src/chromeServices/background/index.ts:88]: Make a const for this key

File: packages/extension/src/App.tsx [hunk 3]
```diff
@@ -67,171 +78,177 @@ const App: FC = () => {
...
-      } else if (search.includes(PARAMETER_TRANSACTION_CANCEL_OFFER)) {
-        handleTransaction({
-          app: GEM_WALLET,
-          type: 'RECEIVE_CANCEL_OFFER/V3',
-          payload: defaultPayload
-        });
+  useBeforeUnload(async () => {
+    // Only sends reject message if there is a transaction in progress, which means the user has not confirmed the
+    // transaction but has closed the extension
+    // const hasTxInProgress = await loadFromChromeSessionStorage('hasTxInProgress', true);
```
Comment (ThibautBremand) [inline packages/extension/src/App.tsx:84]: To be removed

File: packages/extension/src/App.tsx [hunk 4]
```diff
@@ -48,12 +54,17 @@ const App: FC = () => {
...
+  useEffect(() => {
+    setTransactionProgress(TransactionProgressStatuses.IN_PROGRESS);
+  }, [setTransactionProgress]);
 
   const handleTransaction = useCallback(
     (payload: unknown) => {
       if (process.env.NODE_ENV === 'production') {
         chrome.runtime
-          .sendMessage(payload)
+          .sendMessage(chrome.runtime.id, payload)
```
Comment (ThibautBremand) [inline packages/extension/src/App.tsx:67]: To be checked

File: packages/extension/src/contexts/index.ts (hunk 1 above)
Comment (ThibautBremand) [reply packages/extension/src/contexts/index.ts]: Done

**Pull request merged**